react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+function TopicListItem(props) {
+  const { topic, detail } = props.topics;
+
+  return (
+    <li>
+      <div className="list-container">
+        <h2 onClick={props.onClick}>{topic}</h2>
+        { topic === props.state.topic && props.state.view === true &&
+          <p>{detail}</p>
+        }
+      </div>
+    </li>
+  );
+}
+
+class TopicList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      topic: '',
+      view: false
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(event) {
+    const currentTopic = event.target.textContent;
+
+    if (currentTopic !== this.state.topic) {
+      this.setState(state => ({
+        topic: currentTopic,
+        view: true
+      }));
```

Comment:
Instead of storing both the topic and a boolean, you can store only the topic (or even just its id).